File: services/ai-review-service/app/metrics.py
"""
Prometheus metrics for AI review service
"""

from prometheus_client import Counter, Histogram, generate_latest, CONTENT_TYPE_LATEST
import time

# Counter for total reviews processed
reviews_total = Counter(
    'reviews_total',
    'Total number of code reviews processed',
    ['status']  # status: success, error
)

# Histogram for AI service latency (LLM call time)
ai_latency_ms = Histogram(
    'ai_latency_ms',
    'AI service call latency in milliseconds',
    buckets=[100, 500, 1000, 2000, 5000, 10000, 30000]  # buckets in ms
)

def get_metrics():
    """Return Prometheus metrics in text format"""
    return generate_latest()

File: services/orchestrator-service/src/metrics.ts
/**
 * Prometheus metrics for orchestrator-service
 */

import { Registry, Counter, Histogram } from 'prom-client';

// Create a registry to register metrics
export const register = new Registry();

// Counter for total reviews processed
export const reviewsTotal = new Counter({
  name: 'reviews_total',
  help: 'Total number of code reviews processed',
  labelNames: ['status'], // status: done, failed, running
  registers: [register]
});

// Histogram for AI service latency
export const aiLatencyMs = new Histogram({
  name: 'ai_latency_ms',
  help: 'AI service call latency in milliseconds',
  buckets: [100, 500, 1000, 2000, 5000, 10000, 30000], // buckets in ms
  registers: [register]
});

// Register default metrics (CPU, memory, etc.)
register.setDefaultLabels({
  service: 'orchestrator-service'
});

